cap/modules/schemas/models/templates.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of CERN Analysis Preservation Framework.
+# Copyright (C) 2016 CERN.
+#
+# CERN Analysis Preservation Framework is free software; you can redistribute
+# it and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# CERN Analysis Preservation Framework is distributed in the hope that it will
+# be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Analysis Preservation Framework; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+"""Models for schema templates."""
+
+from datetime import datetime
+
+from invenio_db import db
+from invenio_search import current_search_client as es
+from sqlalchemy import event
+from sqlalchemy.orm import validates
+
+from cap.types import json_type
+
+from cap.modules.schemas.serializers import schema_template_serializer
+from cap.modules.schemas.utils import create_index
+
+
+class Template(db.Model):
+    """Model defining analysis ES template mappings."""
+
+    id = db.Column(db.Integer, primary_key=True)
+    prefix = db.Column(db.String(128), unique=False, nullable=False)
+    name = db.Column(db.String(128), unique=False, nullable=False)
+    mapping = db.Column(json_type, default=lambda: dict(), nullable=True)
+
+    created = db.Column(db.DateTime(), default=datetime.utcnow, nullable=False)
+    updated = db.Column(db.DateTime(), default=datetime.utcnow, nullable=False)
+
+    __tablename__ = 'schemas_template'
+
+    def __init__(self, *args, **kwargs):
+        """Possible to set version from string."""
+        version = kwargs.pop('version', None)
+        if version:
+            self.version = version
+
+        super().__init__(*args, **kwargs)
+
+    def __str__(self):
+        """Stringify template object."""
+        return f'{self.prefix}-{self.name}-v{self.version}'
```

Comment:
what kind of value you had on your mind?
would be nice to have either a name under which we can find it in elastic or in our db (unique)
also in your example  if version is `None` will return `prefix-name-v`